Delete old CV versions (#4869)
theforeman/foreman-documentation: guides/common/modules/proc_purging-content-view-versions-by-using-cli.adoc

Reviewer: It isn't clear how this is different from deleting CVVs.

Author: * deleting a CV version: users select a version that they want to delete.
* purging CV versions: you tell Foreman+Katello how many that you want to keep; the rest are deleted if they're not part of a LE.

Author: Clarified by using the abstract of the concept module.

:_mod-docs-content-type: PROCEDURE

[id="purging-content-view-versions-by-using-cli"]
= Purging content view versions by using Hammer CLI

[role="_abstract"]
By default, {Project} keeps every version of a content view.

> You can use Hammer CLI to purge excess numbers of content view versions on {Project} to limit disk space usage.

Reviewer: ```suggestion
You can purge excess numbers of content view versions on {Project} to limit disk space usage.
```
I think it's unnecessary to repeat information that is in the heading. Your call.

Reviewer: And can we, please, move the first sentence after the second one?
```suggestion
You can purge excess numbers of content view versions on {Project} to limit disk space usage.
By default, {Project} keeps every version of a content view.
```

Author: Applied both suggestions.

> You can use Hammer CLI to restrict the number of kept content view versions on {Project} to limit disk space usage.

Reviewer: This sounds like a description of a setting, but that's not the case.

Author: Slightly reworded.

> Removing content view versions is a tradeoff between disk space and possibly bandwidth during content synchronizations to {SmartProxyServers} and the possibility to roll back to older content snapshots.

Reviewer: ```suggestion
Removing content view versions is a tradeoff between disk space and possibly bandwidth during content synchronizations to {SmartProxyServers} and the possibility to roll back to older versions.
```
I'm uncertain about calling it "snapshots". Your call.

Reviewer: I've just seen "snapshots" in other parts of the CV docs. It should be okay.

> Hammer only considers deleting content view versions that are not part of any lifecycle environment.

Reviewer 2: nit: It's the Foreman application (specifically the Katello plugin) that does the "considering" logic. Hammer is very simple; it only translates CLI params into API requests and back.

Author: Thanks, fixed.

. Optional: View your content view versions:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----
$ hammer content-view info \
--name "_My_Content_View_Name_" \
--organization-id _My_Organization_ID_
----
. Clean up content view versions:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----
$ hammer content-view purge \
--name "_My_Content_View_Name_" \
--organization-id _My_Organization_ID_ \
--versions-to-keep _My_Number_Of_Versions_To_Keep_
----
+

> Hammer only considers deleting content view versions that are not part of any lifecycle environment.

Reviewer: This should go above the procedure.

Author: Moved.
